src/compiler/scala/tools/nsc/transform/Delambdafy.scala

```diff
@@ -294,12 +300,25 @@ abstract class Delambdafy extends Transform with TypingTransformers with ast.Tre
       case Template(_, _, _) =>
         def pretransform(tree: Tree): Tree = tree match {
           case dd: DefDef if dd.symbol.isDelambdafyTarget =>
-            if (!dd.symbol.hasFlag(STATIC) && methodReferencesThis(dd.symbol)) {
+            val ddef = if (!dd.symbol.hasFlag(STATIC) && methodReferencesThis(dd.symbol)) {
               gen.mkStatic(dd, dd.symbol.name, sym => sym)
             } else {
               dd.symbol.setFlag(STATIC)
               dd
             }
+            if (settings.Ydelambdafy.value == "method-ref")
+              ddef match {
+                case DefDef(_, _, _, List(self :: Nil), _, Apply(select @ Select(core: Ident, _), Nil))
+                  if core.symbol == self.symbol
+                    && select.symbol.isMethod
+                    && { val x = select.symbol.owner; x != ArrayClass && !x.isPrimitiveValueClass && !x.isDerivedValueClass }
```

Comment:
hmm, makes me wonder what else we should exclude. maybe some "fake" `j.l.Object` members (eq, synchronized), `String.+`? https://github.com/scala/scala/blob/0bfbfe7e0a57e92e927f4c1333884da07e25db99/src/compiler/scala/tools/nsc/backend/ScalaPrimitives.scala#L212

Reply:
Turns out not to be necessary.  I think it used to be necessary before I added the `target.removeAttachment[JustMethodReference]` part and fixed the details in the backend.

After taking this whole clause the whole suite passes.

Reply:
Actually, I wrote a comprehensive test suite against those fake methods and I need to check for (non-derived) primitives.